### PR TITLE
Create a class to complement Song.java

### PR DESCRIPTION
**Pull request by RazvanBudacaTechquarter**
This class will be used to store all the information taken from the server, then we are going to use it to store info about the most searched songs.

### CODE REVIEW

File: src/main/java/com/google/alpollo/SongInfo.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+/**
+ * Class used for storing data related to the Natural Language API analysis.
+ * Extending the Song class, it provides all the info related to a specific song.
+ * 
+ * Score tells us the overall positivty of the song, ranging from -1.0 to 1.0.
+ * Magnitude tells us how strong the main sentiment of the song is, ranging from 0.0 to +inf.
+ * TopSalientEntities is a list containing the 10 most important words, given the song context.
+ */
+public class SongInfo extends Song{
```
Comment (jeinygroove) [inline src/main/java/com/google/alpollo/SongInfo.java]: space before `{`

File: src/main/java/com/google/alpollo/SongInfo.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package com.google.alpollo;
+
+import java.util.List;
+import com.google.alpollo.model.SongEntity;
+
+/**
+ * Class used for storing data related to the Natural Language API analysis.
+ * Extending the Song class, it provides all the info related to a specific song.
+ * 
+ * Score tells us the overall positivty of the song, ranging from -1.0 to 1.0.
```
Comment (jeinygroove) [inline src/main/java/com/google/alpollo/SongInfo.java]: move these comments to fields

File: src/main/java/com/google/alpollo/SongInfo.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package com.google.alpollo;
+
+import java.util.List;
+import com.google.alpollo.model.SongEntity;
```
Comment (jeinygroove) [inline src/main/java/com/google/alpollo/SongInfo.java:4]: SongEntity doesn't exist in this branch, code doesn't compile

**Review by jeinygroove — COMMENTED** [review]
Seems like you wanted to create this branch from `BudiAnalysis`, but it's not

File: src/main/java/com/google/alpollo/SongInfo.java (hunk 3 above)
Comment (RazvanBudacaTechquarter) [reply src/main/java/com/google/alpollo/SongInfo.java:4]: Totally true, I should have branched from BudiEntityAnalisis, but everything should be fine if we merge that first.

File: src/main/java/com/google/alpollo/SongInfo.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+/**
+ * Class used for storing data related to the Natural Language API analysis.
+ * Extending the Song class, it provides all the info related to a specific song.
+ */
+public class SongInfo extends Song {
+  /** Score tells us the overall positivty of the song, ranging from -1.0 to 1.0.*/
```
Comment (jeinygroove) [inline src/main/java/com/google/alpollo/SongInfo.java:11]: it can be just `//`

File: src/main/java/com/google/alpollo/SongInfo.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  private final float score;
+  /** Magnitude tells us how strong the main sentiment of the song is, ranging from 0.0 to +inf. */
```
Comment (jeinygroove) [inline src/main/java/com/google/alpollo/SongInfo.java:13]: it can be just `//`

File: src/main/java/com/google/alpollo/SongInfo.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  private final float magnitude;
+  /** TopSalientEntities is a list containing the 10 most important words, given the song context.*/
```
Comment (jeinygroove) [inline src/main/java/com/google/alpollo/SongInfo.java:15]: it can be just `//`

File: src/main/java/com/google/alpollo/SongInfo.java (hunk 4 above)
Comment (RazvanBudacaTechquarter) [reply src/main/java/com/google/alpollo/SongInfo.java:11]: Hmm, the guys told me before I should use javadoc

**Comment by jeinygroove** [comment]
And I still don't see `SongEntity`, please upload changes from `master` then. It's not good when code in your branch doesn't compile, for example, if you will merge this branch to master, it can stop compiling as well.

It's good that you already have `BudiAnalysis` merged to `master`, but if it wasn't, we could have the broken `master` after this merge :(

**Review by jeinygroove — APPROVED** [review]

File: src/main/java/com/google/alpollo/SongInfo.java (hunk 4 above)
Comment (jeinygroove) [reply src/main/java/com/google/alpollo/SongInfo.java:11]: OK

**Review by jgaillard85 — COMMENTED** [review]
I am usually all for short PRs, but here you create a new class without using it. That makes it hard to see whether this is good or not. For example, I don't know at the moment why you are adding this class, whether having this as a subclass of Song is the best approach...

**Comment by RazvanBudacaTechquarter** [comment]
> I am usually all for short PRs, but here you create a new class without using it. That makes it hard to see whether this is good or not. For example, I don't know at the moment why you are adding this class, whether having this as a subclass of Song is the best approach...

I agree with you, this class was created while keeping in mind future plans. Maybe it's better to leave it for later.

**Comment by RazvanBudacaTechquarter** [comment]
> And I still don't see `SongEntity`, please upload changes from `master` then. It's not good when code in your branch doesn't compile, for example, if you will merge this branch to master, it can stop compiling as well.
> 
> It's good that you already have `BudiAnalysis` merged to `master`, but if it wasn't, we could have the broken `master` after this merge :(

SongEntity is part of the BudiEntityAnalysis branch, which is not yet merged. This feature should have been branched from BudiEntityAnalysis.

**Pull request closed without merging**